File: 프로그래머스/2/12911. 다음 큰 숫자/다음 큰 숫자.py
def solution(n):
    answer = 0
    s = bin(n)[2:]
    one_len = ""
    for i in s:
        if i == "1":
            one_len += i
    while True:
        n += 1
        s = bin(n)[2:]
        compet = ""
        for i in s:
            if i == "1":
                compet += i
        if len(compet) == len(one_len):
            answer = n
            break
    return answer

File: Java/프로그래머스/2/12911. 다음 큰 숫자/다음 큰 숫자.java
class Solution {
    public int solution(int n) {
        int answer = 0;
        int nextN = n;
//         비트 수 세기
        int nCount = Integer.bitCount(n);
        while(true){
//             조건 1
            nextN++;
//             조건 2
            int nextNCount = Integer.bitCount(nextN);
            if(nextNCount == nCount) break;   
        }
        return nextN;
    }
}